dfusion_rust_core/src/database/mod.rs

```diff
@@ -1,11 +1,13 @@
 mod error;
+mod graph_reader;
 
 use web3::types::{H256, U256};
 
 pub use error::*;
+pub use graph_reader::GraphReader;
 use super::models;
 
-pub trait DbInterface {
+pub trait DbInterface: Send + Sync + 'static {
```

Comment:
What is the reason we need the Send etc here?

Reply:
DbInterface is a field on `FluxTransitionHandler`, which implements `EventHandler`, which itself is `Send + Sync + Debug`. Any field on such a datatype needs to be `Send + Sync` itself.

`EventHandler` is `Send + Sync` since it's a field on the RuntimeHost, which implements `pub trait RuntimeHost: Send + Sync + Debug`.

The static is actually unnecessary, good question/catch!